tests/test_dle.py

```diff
@@ -489,7 +489,9 @@ def test_collapsing(self, eqns, expected, blocking):
             op = Operator(eqns, opt=('blocking', 'simd', 'openmp',
                                      {'blockinner': True, 'par-collapse-ncores': 1,
                                       'par-collapse-work': 0}))
-            iterations = FindNodes(Iteration).visit(op._func_table['bf0'])
+            iterations = FindNodes(Iteration).visit(op)
+            assert len(retrieve_iteration_tree(op)) == 1
```

Comment:
you don't need `iterations` anymore if you call `retrieve_iteration_tree` and check that there's only one tree, because then such tree, say `trees[0]`, is == `iterations`. This applies here and below...

Reply:
yes